test-adapter/_testadapter/tink_unittest/Injector.hx

```diff
@@ -13,8 +13,10 @@ class Injector {
 		for (field in fields) {
 			switch (field.name) {
 				case "shouldRun":
-					field.patch(Start, macro if (!_testadapter.data.TestFilter.shouldRunTest($v{Macro.filters}, this.suite.info.name, this.info.name))
-						return false);
+					field.patch(Start,
+						macro if (!_testadapter.data.TestFilter.shouldRunTest($v{Macro.filters},
+								this.suite.info.name + ' [' + this.suite.info.pos.fileName + ':' + this.suite.info.pos.lineNumber + ']', this.info.name))
```

Comment:
Changing the name to make unique identifier probably helps solve a problem you have on your side, however currently test adapter extension (VSCode part) uses className for both label and id when reporting to Test Explorer UI extension. 
When you run tests with a filter then we just receive a list of ids to run.

Technically we could change class id to something else, but usually a class name with full package info is sufficient.

For individual tests we generate an id from className and testName, by simple concatenation:  className + `.` + testName. That's why we can filter tests because usually they know their names.

We have full control over the ids we report to Test Explorer UI extension, but changing anything will effect all test frameworks.

Reply:
I believe this library should support the more generic tink_testrunner, instead of tink_unittest which is just a specific implementation of the interfaces defined in tink_testrunner. And actually it just happens doing so already. Because the injected build macro really only affects the `tink.testrunner` package.

The problem is that in tink_testrunner, a `Suite` is essentially just an array of cases + a name, the simplest representation would be:
```haxe
typedef Suite = {
  name:String,
  cases:Array<Case>,
}
```
(A more detailed explanation can be found [here](https://haxetink.github.io/tink_testrunner/#/getting-started/overview))

Thus it is not necessary to create a new class (e.g. `class MySuite extends Suite`) in order to hold some tests. The only way to identify a Suite is by its name, and, as a result, there can be more than one Suite of the same name, but that is users' choice.

Back to the issue. In order to really identify two Suites of the same name is to use the position of their declaration:

```haxe
// in the following case the constructor will store the PosInfos of the call.
// So that we can later use such info to distinguish the two same-named Suite.
var suite1 = new Suite('MySuiteName', [/*... cases ...*/]); 
var suite2 = new Suite('MySuiteName', [/*... cases ...*/]);

abstract Suite({name:String, cases:Array<Case>, pos:PosInfos}) {
  public function new(name, cases, ?pos:PosInfos) {...}
}
```

Using suite name + position as the "className" apparently allows me to distinguish two suites of the same name (name="Outcomes")
<img width="345" alt="screen shot 2019-02-24 at 7 37 13 pm" src="https://user-images.githubusercontent.com/1103788/53298627-9b51db80-386b-11e9-8078-f8fb3a28c214.png">

Reply:
As for tink_testrunner <-> tink_unittest there were some discussions on what name to use. We settled on tink_unittest (even though we hook into tink_testrunner), because that is what people use in their projects. 
If there was a second implementation of a unittest framework building upon tink_testrunner, we would probably support it out of the box, without any changes. I'm not aware of any, but I'm not really fluent in the tink_ eco system.

You are right that Suite names might not always be unique, e.g. for buddy nobody is stopping you from using the same description multiple times. Test adapter currently doesn't handle that, so we might need to rework class / suite ids to include position info. But that probably requires a refactoring of all integrations.

Your screenshot also shows the issue I was talking about earlier. tink_testrunner currently cannot differentiate between tests disabled by users and filtered tests. So once you start a filtered run, you loose all existing results and they report as skipped, even though they weren't from user's perspective. When you run one Suite or one Case only those results should be reported, leaving other results untouched. 
I mean you will probably start a full, unfiltered test run after fixing one failing test, to see if there are any side effects, but that should be user's choice not a requirement.